src/bot.ml

```diff
@@ -192,6 +192,63 @@ let callback _conn req body =
             |> Lwt.async ;
             Server.respond_string ~status:`OK ~body:"Handling minimization." ()
             )
+          else if
+            string_match
+              ~regexp:
+                (f "@%s:? [Cc][Ii][- ][Mm]inimize:?\\( [^\n]*\\)?" bot_name)
+              body
+          then (
+            let requests =
+              String.split ~on:' '
+                (Stdlib.String.trim
+                   (* FIXME why not `String.trim`? ocamlc complains it's unbound... (cf https://coq.zulipchat.com/#narrow/stream/243318-coqbot-devs.20.26.20users/topic/.60String.2Etrim.60/near/236752334) *)
+                   (Str.global_replace (Str.regexp "[ ,]+") " "
+                      (Str.matched_group 1 body)))
+            in
+            match comment_info.pull_request with
+            | Some pull_request ->
+                (fun () ->
+                  (* XXX FIXME do we want this???
+                     action_as_github_app ~bot_info ~key ~app_id
+                       ~owner:comment_info.issue.issue.owner
+                       ~repo:comment_info.issue.issue.repo
+                  *)
+                  minimize_failed_tests ~bot_info
+                    ~owner:comment_info.issue.issue.owner
+                    ~repo:comment_info.issue.issue.repo
+                    ~base:pull_request.base.sha ~head:pull_request.head.sha
+                    ~pr_number:(Some comment_info.issue.number)
+                    ~head_pipeline_summary:None
+                    ~request:
+                      ( match requests with
+                      | [] ->
+                          RequestSuggested
+                      | ["all"] ->
+                          RequestAll
+                      | _ ->
+                          RequestExplicit requests ))
+                |> Lwt.async ;
+                Server.respond_string ~status:`OK
+                  ~body:"Handling CI minimization." ()
+            | None ->
+                (fun () ->
+                  (*XXX DO we need this? action_as_github_app ~bot_info ~key ~app_id
+                    ~owner:comment_info.issue.issue.owner ~repo:comment_info.issue.issue.repo*)
+                  GitHub_mutations.post_comment
+                    ~id:
+                      comment_info.id
+                      (* XXX This was comment_thread_id, is id correct here? *)
+                    ~message:
+                      (f
+                         "Hey @%s, you cannot run CI minimization on issues \
+                          that are not pull requests."
+                         comment_info.author)
+                    ~bot_info
+                  >>= GitHub_mutations.report_on_posting_comment)
+                |> Lwt.async ;
+                (* XXX IS this an Ok or a failure? *)
```

Comment:
What about moving all this logic to a function in `action.ml`? This would also mean that you don't need to export `ci_minimization_request` in `action.mli`.

Reply:
I'm not sure what logic you're suggesting that I move, nor what the conceptual specification is.  If you're just suggesting that I move the request-parsing logic, I'm inclined to just export a separate parsing function and make the `ci_minimization_request` type opaque.  Otherwise the "logic" here is really just passing through a bunch of arguments.

Reply:
It's in particular the `match comment_info.pull_request` and the `GitHub_mutations.post_comment` called directly in this file that I'm uncomfortable with.

Reply:
Moved.  I've called the function `ci_minimize` and pass it `~bot_info ~comment_info ~requests`.  Let me know if this is good